src/ios/CDVCamera.m

```diff
@@ -402,6 +444,115 @@ - (NSString*)tempFilePath:(NSString*)extension
     return filePath;
 }
 
+// This is not absolute, but more mightNeedOrientationCorrection
+- (BOOL) needsOrientationCorrection:(UIImage*)image options:(CDVPictureOptions*)options
+{
+    // TODO use image to detect if it needs an orientation correction
+
+    // See FIXME #2
+    //    if (options.sourceType != UIImagePickerControllerSourceTypeCamera) {
+    //        return false;
+    //    }
+    return options.correctOrientation;
+}
+
+- (BOOL) needsResize:(CDVPictureOptions*)options
+{
+    return (options.targetSize.height > 0 && options.targetSize.width > 0);
+}
+
+- (BOOL) needsEdit:(UIImage*)image options:(CDVPictureOptions*)options
+{
+    return [self needsOrientationCorrection:image options:options] || [self needsResize:options];
+}
+
+- (BOOL) needsSavingToPhotoAlbum:(UIImage*)image options:(CDVPictureOptions*)options
+{
```

Comment:
If the user specifies the option saveToPhotoAlbum and the image already exists in the PhotoLibrary, shouldn't we save it to the SavedPhotoAlbum ? (since SavedPhotoAlbum is different from the PhotoLibrary).